geoutils/georaster.py

```diff
@@ -194,6 +194,27 @@ def _read_attrs(self, attrs=None):
         for attr in attrs:
             setattr(self, attr, getattr(self.ds, attr))
 
+    def set_data(self, new_data):
```

Comment:
Something to be tested (I just found about these functions), but I think we should use setter and getter decorators like in the last example here: https://www.geeksforgeeks.org/getter-and-setter-in-python/

Reply:
yes, exactly my thinking too.

Reply:
Some quick example I've tested:
```
class Raster:
    def __init__(self, data=None):
        self.shape = (3,3)
        if data is None:
            self._data = np.arange(9).reshape((3,3))
        else:
            self.data = data

    @property
    def data(self):
        return self._data
        
    @data.setter
    def data(self, d):
        if d.shape != self.shape:
            raise ValueError("Data has wrong shape")
        else:
            self._data = d
```
Then
```
img = Raster()
img.data
```
array([[0, 1, 2],
       [3, 4, 5],
       [6, 7, 8]])
```
img.data = np.arange(9)
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-27-e54a5000bc4c> in <module>
----> 1 img.data = np.arange(9)

<ipython-input-19-cb9ce1fef3a3> in data(self, d)
     14     def data(self, d):
     15         if d.shape != self.shape:
---> 16             raise ValueError("Data has wrong shape")
     17         else:
     18             self._data = d

ValueError: Data has wrong shape